jenkins/job-configs/kubernetes-jenkins/kubernetes-e2e-kops-aws.yaml

```diff
@@ -0,0 +1,67 @@
+- job-template:
+    name: 'kubernetes-e2e-kops-{suffix}'
+    node: '{jenkins_node}'
+    description: '{description} Test owner: {test-owner}.'
+    jenkins_node: 'e2e'
+    disabled: '{obj:disable_job}'
+    properties:
+        - build-discarder:
+            days-to-keep: 7
+    triggers:
+        - reverse:
+            jobs: '{trigger-job}'
+            result: success
+        - timed: '{cron-string}'
+    publishers:
+        - junit-publisher
+        - email-ext:
+            recipients: '{emails}'
+        - gcs-uploader
+    provider-env: |
+        export E2E_OPT="--deployment kops --kops $WORKSPACE/kops -kops-cluster {suffix}.test-aws.k8s.io -kops-state s3://k8s-kops-jenkins/"
+        export AWS_CONFIG_FILE=/workspace/.aws/credentials
+        # This is needed to be able to create PD from the e2e test
+        export AWS_SHARED_CREDENTIALS_FILE=/workspace/.aws/credentials
+        export KUBE_SSH_USER=admin
+    builders:
+        - activate-gce-service-account
+        - shell: |
+            {provider-env}
+            {job-env}
+            {post-env}
+            if ! curl -fsS --retry 3 -o "$WORKSPACE/kops.url" "https://storage.googleapis.com/kops-ci/bin/latest-ci.txt"; then
```

Comment:
We should talk about this. I think it defeats the purpose. To most of us writing these jobs, you've got the `yaml`, which is super transparent, you've got `e2e-runner.sh`, which is anything but transparent, and then you hit `e2e.go`. This is a super minor setup/download, all things considered, but you didn't want it in `e2e.go`, where it would be obvious and easy to maintain in `go`. It's *not* easy to maintain in `e2e-runner.sh`, which is actually the next most obvious place to put it, because I need to plumb an env variable through. It's not easy to maintain in a separate binary, because I end up writing `yaml` code that looks like this:
```
if ! kops-download.sh $WORKSPACE/kops $WORKSPACE/kops.env; then
  echo "Can't retrieve kops" >&2
  exit 1
fi
. $WORKSPACE/kops.env
```

That's about two lines less complicated than the `yaml` code I wrote already, which is only noisy because of the rigorous error checking that a *ton* of the other `yaml` shell code *doesn't do*. Furthermore, neither case is dockerized, which is kind of bad, but not the worst thing in the world. So, in conclusion, I think this should either stay in the `yaml` or move down into `e2e.go`, and not get lost in some inscrutable morass in the middle.